NeoBurgerTestnetVote.cs

```diff
@@ -0,0 +1,134 @@
+        public static Iterator GetVotersOfProposal(BigInteger proposal_id) => new StorageMap(Storage.CurrentContext, (ByteString)new byte[] { PREFIX_VOTE } + (ByteString)proposal_id).Find((FindOptions)((byte)FindOptions.KeysOnly + (byte)FindOptions.RemovePrefix));
+        public static UInt256 ProposalDigest(UInt160 scripthash, string method, object[] args, BigInteger nonce) => (UInt256)CryptoLib.Sha256(StdLib.Serialize(new object[] { scripthash, method, args, nonce }));
+
+        struct ProposalAttributesStruct
+        {
+            public BigInteger id;
+            public UInt160 provider;
+            public UInt160 scripthash;
+            public string method;
+            public ByteString[] args;
+            public BigInteger nonce;
+            public BigInteger created_time;
+            public BigInteger voting_deadline;
+        }
+
+        public static void _deploy(object data, bool update)
+        {
+            StorageMap proposal_id_map = new(Storage.CurrentContext, new byte[] { PREFIX_PROPOSAL });
+            ProposalAttributesStruct proposal_attributes = new();
+            proposal_attributes.id = 0;
+            proposal_id_map.PutObject((ByteString)(BigInteger)0, proposal_attributes);
+        }
+
+        public static BigInteger NewProposal(UInt160 provider, BigInteger proposal_id, UInt160 scripthash, string method, ByteString[] args)
+        {
+            ExecutionEngine.Assert(Runtime.CheckWitness(provider));
+            StorageMap proposal_id_map = new(Storage.CurrentContext, new byte[] { PREFIX_PROPOSAL });
+            if ((BigInteger)proposal_id_map.Get((ByteString)proposal_id) != 0 || ((ProposalAttributesStruct)proposal_id_map.GetObject((ByteString)(proposal_id - 1))).id != proposal_id - 1)
+                throw new Exception("Invalid proposal id");
+
+            Storage.Put(Storage.CurrentContext, new byte[] { PREFIX_PROPOSAL_LATEST_ID }, (ByteString)proposal_id);
+            
+            ProposalAttributesStruct proposal_attributes = new();
+            proposal_attributes.id = proposal_id;
+            proposal_attributes.provider = provider;
+            proposal_attributes.scripthash = scripthash;
+            proposal_attributes.method = method;
+            proposal_attributes.args = args;
+            proposal_attributes.nonce = Runtime.GetRandom();
+            BigInteger current_time = Runtime.Time;
+            proposal_attributes.created_time = current_time;
+            proposal_attributes.voting_deadline = current_time + VOTING_PERIOD;
+            proposal_id_map.PutObject((ByteString)proposal_id, proposal_attributes);
+
+            return proposal_id;
+        }
+
+        public static void Delegate(UInt160 from, UInt160 to)
+        {
+            ExecutionEngine.Assert(Runtime.CheckWitness(from));
+            StorageMap delegate_map = new(Storage.CurrentContext, PREFIX_DELEGATE);
+            if (to == UInt160.Zero || to == from)
+                delegate_map.Delete(from);
```

Comment:
IMO, keeping the to as Zero or itself will be better since it can save GAS for next Delegate operation.

Reply:
Keeping as zero adds difficulty in collection delegation relationship information. This contract will be deployed on testnet, where GAS consumption is not a problem.

Reply:
Does it mean that we won't delete value on mainnet?